spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -1489,10 +1489,31 @@ def name_id_version(format_urn)
             expect(response.status).to eq(200)
             expect(@analytics).to have_logged_event('SAML Auth', hash_including(analytics_hash))
           end
+
+          context 'Certificate does not match because of order bug' do
+            let(:encryption_cert_matches_matching_cert) { false }
+            let(:matching_cert_serial) { nil }
+            let(:request_sp) { double  }
+
+            before do
+              service_provider.update(certs: ['sp_sinatra_demo', 'saml_test_sp'])
+              allow_any_instance_of(
+                SamlIdp::ServiceProvider,
+              ).to receive(:matching_cert).and_return nil
```

Comment:
`allow_any_instance_of` is my least favorite way to stub but, this stuff does really deep so I don't have a good alternative to propose at the moment

Reply:
i hate it! but it would have needed a lot of setup that i think would effectively be doing the same thing :(